### DISCUSSION TITLE
Docs for api endpoints

### DISCUSSION THREAD

**Discussion by 8ball030** [TwiN/gatus#451] [Q&A] [answered]
Hi i was just wondering if there are any docs for the endpoints exposed

**Comment by TwiN** [comment, marked as answer]
There's this from the README https://github.com/TwiN/gatus#api, but the rest is only available through the code.

All endpoints are defined here: https://github.com/TwiN/gatus/blob/f6e7e346b68266f6c8bf719f27b452e7e846e441/controller/handler/handler.go#L34-L48

But I assume the one you're interested in is the `/api/v1/endpoints/statuses` endpoint, which returns a slice (array) of EndpointStatus structs, whose definition can be found at https://github.com/TwiN/gatus/blob/f6e7e346b68266f6c8bf719f27b452e7e846e441/core/endpoint_status.go#L6-L28

If you want an actual example of the body, see https://status.twin.sh/api/v1/endpoints/statuses

**Reply by 8ball030** [reply]
Yo,
That was exactly what I needed. Thanks so much for your help. 

This is an amazing project btw. Super light and snappy and love the config language. 

I'm working with a dynamic config map, how would you usually trigger a refresh of the container in this case?

**Reply by TwiN** [reply]
Gatus _should_ automatically pick up the change after 30ish seconds by default.

See https://github.com/TwiN/gatus#reloading-configuration-on-the-fly

It doesn't work on _every_ platform though. Where are you deploying Gatus?

**Reply by 8ball030** [reply]
It however *does* work on mine!

![image](https://user-images.githubusercontent.com/35799987/231714066-f6f372f9-ede7-4bd4-a6fb-3c01beab2141.png)

**Reply by 8ball030** [reply]
running typhoon on aws ec2 boxes
